### DISCUSSION TITLE
Bidirectional relations in the control panel

### DISCUSSION THREAD

**Discussion by mihob** [craftcms/cms#11726] [Q&A]
Is there any way to maintain bidirectional relations in the control panel?

Example:
I have the entries Awards and Persons. Awards are assigned to persons and vice versa. 

It would make it much easier for the editor if he could maintain the relation on both entries.

**Comment by bencroker** [comment]
The control panel allows you to maintain relationships between elements from the source element only. So you can either assign awards to a persons field on the persons edit page, or assign persons to an awards field on the awards edit page. 

In Twig, of course, you can create bi-directional queries, so that you can determine and output reverse relationships between elements. To do so in the control panel, you'd need to use a plugin such as [Reverse Relations](https://plugins.craftcms.com/reverserelations).

**Comment by mihob** [comment]
Thank you, I know that this is possible by query. But I am explicitly interested in the representation in the control panel. This would really improve the maintenance and overview. And it would be much easier to query both sides of relations in graphql.

I have already looked at the plugin Reverse Relations, but graph ql is not supported and therefore does not help me.

**Reply by brandonkelly** [reply]
It does sound like that plugin will work for you.

You can already fetch entries via reverse relations in GraphQL out of the box, so not much the field type could add to make that better.

**Reply by mihob** [reply]
This is true, but only via an additional query.  For example, if my entry "persons" has the entry field "awards", I can query all assigned awards with one query.

```
{
    entries(type: "person"){
        id
        title
        awards {
            id
            title      
        }
    }
}
```

But the other way around, I would have to query the awards first and then issue a single query for each award, which queries the related persons. This is not really performant. Or do i get something wrong?

**Reply by brandonkelly** [reply]
Yeah that’s correct.

There’s an existing discussion around this you can follow here: #8424

**Reply by mihob** [reply]
Thanks!

For now i will give the plugin a try
